subscription/views.py

```diff
@@ -999,41 +1000,73 @@ def post(self, request):
 
 
 ## New APIs for individual 
-# def is_paid_page(uri):
-#     patterns = UriPattern.objects.values_list('pattern', flat=True)
-#     return any(re.match(pattern, uri) for pattern in patterns)
+# def is_premium_page(url):
+#     premium_urls = PremiumUsageUnits.objects.values_list('url', flat=True)
+#     return any(re.match(premium_url, url) for premium_url in premium_urls)
+
+# def get_premium_units(url):
+#     logger.debug("get_premium_units "+url)
+#     try:
+#         premium_page = PremiumUsageUnits.objects.get(url=url)
+#         return premium_page.units_consumed
+#     except PremiumUsageUnits.DoesNotExist:
+#         # If no exact match, try regex matching
+#         for premium_page in PremiumUsageUnits.objects.all():
+#             if re.match(premium_page.url, url):
+#                 return premium_page.units_consumed
+#     return 1  # Default to 1 unit if not found in PremiumUsageUnits
+
```

Comment:
We want to change the method naming/implementation so that it applies to all URL rules (i.e. return 0 if the URL is not a metered one). Implementation would be to first check the UriPattern table (remember you need to check the AccessType as well since it could be login required type), then check in the premium units table. If premium unit, return its units_consumed value, if normal paid uri, return 1, otherwise return 0.